node/src/store/indexed_db_store.rs

```diff
@@ -0,0 +1,550 @@
+use std::cell::RefCell;
+use std::convert::Infallible;
+
+use async_trait::async_trait;
+use celestia_types::hash::Hash;
+use celestia_types::ExtendedHeader;
+use rexie::{Direction, Index, KeyRange, ObjectStore, Rexie, TransactionMode};
+use send_wrapper::SendWrapper;
+use serde::{Deserialize, Serialize};
+use serde_wasm_bindgen::{from_value, to_value};
+use tendermint_proto::Protobuf;
+
+use crate::store::{Result, Store, StoreError};
+
+const DB_VERSION: u32 = 1;
+const HEADER_STORE_NAME: &str = "headers";
+const HASH_INDEX_NAME: &str = "hash";
+const HEIGHT_INDEX_NAME: &str = "height";
+
+#[derive(Debug, Serialize, Deserialize)]
+struct ExtendedHeaderEntry {
+    height: u64,
+    hash: Hash,
+    header: Vec<u8>,
```

Comment:
Can you test if the space is reduced if we use base64? Because if the value is save as a string instead of blob it will take much more space.

Reply:
It is being saved as an array of bytes